core/benches/scheduler.rs

```diff
@@ -0,0 +1,493 @@
+    }
+}
+
+fn bench_prio_graph_scheuler(c: &mut Criterion) {
+    let mut stats = BenchStats::default();
+    let bench_env: BenchEnv<RuntimeTransaction<SanitizedTransaction>> = BenchEnv::new(&mut stats);
+
+    let mut group = c.benchmark_group("bench_prio_graph_scheduler_with_sdk_transactions");
+    group.sample_size(10);
+
+    let txs_builders: Vec<(&str, Box<dyn TestTxsBuilder>)> = vec![
+        ("simple_transfer", Box::new(SimpleTransferTxBuilder)),
+        ("many_transfer", Box::new(ManyTransfersTxBuilder)),
+        ("packed_noop", Box::new(PackedNoopsTxBuilder)),
+    ];
+
+    let conflict_conditions: Vec<(&str, bool)> =
+        vec![("non-conflict", false), ("full-conflict", true)];
+
+    let tx_counts: Vec<(&str, usize)> = vec![
+        ("empty_container", 0),
+        ("full_container", TOTAL_BUFFERED_PACKETS),
+    ];
+
+    for (txs_builder_type, txs_builder) in &txs_builders {
+        for (conflict_condition_type, conflict_condition) in &conflict_conditions {
+            for (tx_count_type, tx_count) in &tx_counts {
+                let bench_name =
+                    format!("{txs_builder_type}/{conflict_condition_type}/{tx_count_type}");
+                group.throughput(Throughput::Elements(*tx_count as u64));
+                group.bench_function(&bench_name, |bencher| {
+                    bencher.iter_custom(|iters| {
+                        let mut execute_time: Duration = std::time::Duration::ZERO;
+                        for _i in 0..iters {
+                            // setup new Scheduler and Container for each iteration of execution
+                            let (scheduler, container) = {
+                                let mut bench_container =
+                                    BenchContainer::new(TOTAL_BUFFERED_PACKETS);
+                                bench_container.fill_container(
+                                    txs_builder
+                                        .build(*tx_count, *conflict_condition)
+                                        .into_iter(),
+                                );
+                                let scheduler = PrioGraphScheduler::new(
+                                    bench_env.consume_work_senders.clone(),
+                                    bench_env.finished_consume_work_receiver.clone(),
+                                    PrioGraphSchedulerConfig::default(),
+                                );
+                                (scheduler, bench_container.container)
+                            };
+
+                            // execute with custom timing
+                            let start = Instant::now();
+                            {
+                                bench_env.run(
+                                    black_box(scheduler),
+                                    black_box(container),
+                                    &mut stats,
+                                );
```

Comment:
this drops the scheduler and the container and that will be included in the time. The point of doing the custom iteration is to avoid including those times. I think we just make `run` take mutable references.

Reply:
@ 4328c93ec8601c1d6e8b09344c5143aac621fee3